test/state/state.cpp

```diff
@@ -497,6 +510,9 @@ TransactionReceipt transition(const StateView& state_view, const BlockInfo& bloc
     gas_used -= refund;
     assert(gas_used > 0);
 
+    // EIP-7623: The gas used by the transaction must be at least the min_gas_cost.
+    gas_used = std::max(gas_used, tx_props.min_gas_cost);
```

Comment:
So in the end gas used may be higher than transaction gas limit? wild

Reply:
Ah no, I guess not, because it's checked in transaction validity

Reply:
Hm in EIP max operation left side is 
```STANDARD_TOKEN_COST * tokens_in_calldata
+ execution_gas_used
+ isContractCreation * (32000 + INITCODE_WORD_COST * words(calldata))
```
but here it is, I think
```
21000
+ STANDARD_TOKEN_COST * tokens_in_calldata
+ execution_gas_used
+ isContractCreation * (32000 + INITCODE_WORD_COST * words(calldata))
```
because `21000` was included in `intrinsic_cost`?

Reply:
By the Yellow Paper and the names we used so far the intrinsic gas includes the base cost of 21000. The EIP extracts the base cost outside of the max function: `21000 + max(intrinsic_without_base, floor_gas)`.
Because also the EIP don't name any of these values, I decided to include the base cost in the `min_gas_cost`. It is easier to use.

Reply:
Oh I see